ex2_hal/athena/equipment_handler/source/housekeeping_athena.c

```diff
@@ -1,141 +1,181 @@
+    uint32_t OBC_uptime_32 = OBC_ticks / 1000;
+    if (OBC_uptime_32 > 655350) {
+        OBC_uptime_32 = 655350;
+    }
+    // Seconds = value*10. Max = 655350 seconds (7.6 days)
+    OBC_uptime_32 = OBC_uptime_32 / 10;
+    // convert OBC_uptime from 32 bit to 16 bit
+    uint16_t OBC_uptime = (OBC_uptime_32 & 255);
+
+    return OBC_uptime;
+}
+
+/**
+ * @brief
+ *      Getter to Athena solar panel supply current
+ * @return
+ * 		  integer solar_panel_supply_curr to store solar panel supply current
+ */
+uint16_t Athena_get_solar_supply_curr() {
+    // insert getter function for solar panel supply current;
+    uint8_t addr = 0b1000101;
+    uint16_t retval;
+    int get_status = ina209_get_current(addr, &retval);
+    return retval;
+}
+
+/**
+ * @brief
+ * 		Athena composition housekeeping getter
+ * @details
+ * 		Contains calls to other functions that each return a portion of Athena
+ *    telemetry data. Generally reports back to housekeeping_service.
+ * @attention
+ * 		New components should follow the structure of the existing components.
+ *    All data needs to be deep copied
+ * @param athena_hk
+ * 		Pointer to struct of athena housekeeping data
+ * @return
+ * 		Last found error will be returned. else no error returned
+ */
+int Athena_getHK(athena_housekeeping *athena_hk) {
+    int temp_status;
+    int return_code = 0;
+
+    /*Add athena HAL housekeeping getters here and put fields in h file
+    create HAL functions here following format of existing
+    also add endianness conversion in Athena_hk_convert_endianness*/
+    temp_status = HAL_get_temp_all(&athena_hk->MCU_core_temp, &athena_hk->converter_temp);
+
+    // Get last 8 digits of the software version
+    memcpy(athena_hk->OBC_software_ver, ex2_hk_version, 8 * sizeof(char));
+
+    // Get OBC uptime: Seconds = value*10. Max = 655360 seconds (7.6 days)
+    athena_hk->OBC_uptime = Athena_get_OBC_uptime();
+
+    // Get boot info
+    boot_info info;
+    if (eeprom_get_boot_info(&info)) {
+        return_code = -1;
+    }
+    athena_hk->boot_cnt = info.count;
+    athena_hk->last_reset_reason = info.reason.swr_reason;
+    athena_hk->boot_src = info.reason.rstsrc;
+
+    // Get solar panel supply current
+    athena_hk->solar_panel_supply_curr = Athena_get_solar_supply_curr();
+
+    if (temp_status != 0)
+        return_code = temp_status;
+
+    // Get solar panel supply current
+    athena_hk->solar_panel_supply_curr = Athena_get_solar_supply_curr();
+
+    // Get SD card usage percentages
+#if defined(HAS_SD_CARD)
+    int32_t iErr;
+    REDSTATFS volstat;
+    iErr = red_statvfs("VOL0:", &volstat);
+    athena_hk->vol0_usage_percent =
+        (uint8_t)((float)(volstat.f_bfree) * 100.0 /
+                  ((float)(volstat.f_blocks))); // assuming block size == sector size = 512B
+    if (iErr == -1) {
+        exit(red_errno);
+    }
+
+#ifdef IS_ATHENA_V2
+    iErr = red_statvfs("VOL1:", &volstat);

```

Comment:
Does VOL1: work here or does it need the trailing /?

Reply:
I just used the path prefix in gaRedVolConf, which didn't have the trailing /, someone would need to test it on athena